modules/publisher/views.py

```diff
@@ -319,56 +319,60 @@ def build_image_info(image, image_type):
 
 
 def post_market_snap(snap_name):
-    snap_id = api.get_snap_id(snap_name, flask.session)
+    changes = loads(flask.request.form['changes'])
 
-    if 'submit_revert' in flask.request.form:
-        flask.flash("All changes reverted.", 'information')
-    else:
+    if changes:
+        snap_id = api.get_snap_id(snap_name, flask.session)
```

Comment:
@tbille I moved this line from above (as it is only needed for screenshots API call), but is it needed at all?
I think we have snap_id from the form? At least we use it later when posting body_json.

Reply:
OK. Done, seems to work as it should.